src/docs/runtime-nominal-type-descriptor.md

```diff
@@ -1,3 +1,65 @@
 # Swift Nominal Type Descriptor
 
-TBD
\ No newline at end of file
+In Swift, descriptors are implicit data types that are separate from type metadata to provide more information about the type.
+Nominal type descriptors in Swift provide information about elements that are common to most nominal types: struct, class and enum. Protocols are **not** included, even though they are technically nominal types. Protocols with associated types are also **not** included as [has been discussed before](binding-pats.md) PATs are not actual types.
+
+Nominal type descriptors contain the following information that we will need at runtime:
```

Comment:
Is there any Apple source which describes the layout of the Nominal Type Descriptor?

Reply:
Yes, [but it is marked as horribly out of date](https://github.com/swiftlang/swift/blob/c0a55e11d4309a34ef2612c91ca21453ccd5e721/docs/ABI/TypeMetadata.rst#nominal-type-descriptor).
The information I use is from the source.